omnigibson/prims/entity_prim.py

```diff
@@ -237,51 +225,31 @@ def update_joints(self):
         self._joints = dict()
         self.update_handles()
 
-        # Handle case separately based on whether the handle is valid (i.e.: whether we are actually articulated or not)
-        if (not self.kinematic_only) and self._handle is not None:
-            root_prim = get_prim_at_path(self._dc.get_rigid_body_path(self._root_handle))
-            n_dof = self._dc.get_articulation_dof_count(self._handle)
+        # Handle case separately based on whether we are actually articulated or not
+        if not self.kinematic_only:
+            self._n_dof = self._articulation_view.num_dof
 
+            # TODO: This still includes fixed joints for objects that also have non-fixed?
```

Comment:
what is this TODO saying exactly? sorry i'm a bit confused

Reply:
nvm I fixed this issue later.